app/src/main/java/io/intrepid/contest/screens/contestjudging/scoreentrydetail/RateEntryPresenter.java

```diff
@@ -0,0 +1,80 @@
+package io.intrepid.contest.screens.contestjudging.scoreentrydetail;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+
+import java.util.List;
+
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Entry;
+import io.intrepid.contest.models.Score;
+import io.intrepid.contest.screens.contestjudging.scoreentrydetail.expandablerecycler.ExpandableViewHolder;
+
+class RateEntryPresenter extends BasePresenter<RateEntryContract.View> implements RateEntryContract.Presenter {
+    private final Entry entry;
+
+    RateEntryPresenter(@NonNull RateEntryContract.View view,
+                       @NonNull PresenterConfiguration configuration,
+                       Entry entry) {
+        super(view, configuration);
+        this.entry = entry;
+    }
+
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+        view.showEntry(entry);
+    }
+
+
+    @Override
+    public void onCategoriesLoaded(List<Category> categories) {
+        view.showCategories(categories);
+    }
+
+    @Override
+    public void onCategoryRateChanged(ExpandableViewHolder viewHolder, Category category, float newRating) {
+        int scoreIndex = entry.indexOf(category);
+        if (scoreIndex > 0) {
+            entry.getScores().set(scoreIndex, new Score(category.getId(), (int) newRating));
+        } else {
+            entry.getScores().add(new Score(category.getId(), (int) newRating));
```

Comment:
I agree 100% I think I was just trying to get my tests to pass but actually it's a lot easier to test mockEntry.set and mockEntry.add ...